test/acceptance/sniff.test.js

```diff
@@ -65,6 +65,7 @@ test('Should update the connection pool', t => {
     })
 
     // run the sniffer
+    // TODO: modify node roles when master is not supported
```

Comment:
I agree. For the tests, most tests are modified to contains two roles, for example:

```
  // TODO: modify node roles when master is not supported
  t.test('nodesToHost', t => {
    t.test('publish_address as ip address (IPv4)', t => {
      const pool = new ConnectionPool({ Connection })
      const nodes = {
        a1: {
          http: {
            publish_address: '127.0.0.1:9200'
          },
          roles: ['cluster_manager', 'data', 'ingest']
        },
        a2: {
          http: {
            publish_address: '127.0.0.1:9201'
          },
          roles: ['master', 'data', 'ingest']
        }
      }

```

I didn't create the same tests for 'cluster_manager' since the tests will be exactly the same and will bring a lot of duplications. I think there are two conditions:
1) for multiple nodes, each node could have different roles, then we could just modify one node's role, just like the above example
2)if there are some test cases that need to repeat. we will just repeat the test cases for both roles. 

I could re-check all the tests. What do you think?



or do you think

Reply:
sure. will add some tests.